### PR TITLE
[12.0][MIG] mail_cleanup

### PR DESCRIPTION
**Pull request by asaunier**
Ported migration to 11.0 whose PR is available at https://github.com/OCA/server-tools/pull/1411

### CODE REVIEW

**Comment by sebalix** [comment]
@asaunier add `server-env` to a `oca_dependencies.txt` file at the root of the repository (like here: https://github.com/OCA/sale-workflow/blob/12.0/oca_dependencies.txt), as `mail_cleanup` depends on `mail_environment` which is hosted in OCA/server-env now.

**Comment by asaunier** [comment]
@sebalix Thanks a lot for the hint!

Now we have the same code coverage issue than for the 11.0 PR https://github.com/OCA/server-tools/pull/1411 :P

File: .travis.yml [hunk 1]
```diff
@@ -27,6 +27,7 @@ install:
   - git clone --depth=1 https://github.com/OCA/maintainer-quality-tools.git ${HOME}/maintainer-quality-tools
   - export PATH=${HOME}/maintainer-quality-tools/travis:${PATH}
   - travis_install_nightly
+  - printf '[options]\n\nrunning_env = dev\n' > ${HOME}/.openerp_serverrc
```
Comment (vrenaville) [inline .travis.yml:30]: Do we still need thaht ?, .openerp_serverrc seems pretty old for me

Comment (asaunier) [reply .travis.yml:30]: I have added this change at ff29eeb to fix errors reported by Travis at https://travis-ci.org/OCA/server-tools/builds/569737181?utm_source=github_status&utm_medium=notification
```
  File "/home/travis/dependencies/server-env/server_environment/serv_config.py", line 50, in <module>
1599    "The parameter 'running_env' has not be set neither in base config "
1600Exception: The parameter 'running_env' has not be set neither in base config file option -c or in openerprc.
1601We strongly recommend against using the rc file but instead use an explicit config file with this content:
1602[options]
1603running_env = dev
```
I don't remember exactly where but I found this ``.travis.yml`` line in one of C2C's (not that old?) projects and it seems to do the job :)

Comment (asaunier) [reply .travis.yml:30]: I have tried to use ``.odoo_serverrc`` (see 4aa7f03) but not sure it really exists, I have found no occurence of it) or ``.odoorc`` (see e86e9b1). But with no success.

**Review by Tonow-c2c — APPROVED** [review]
Just code review

**Review by damdam-s — APPROVED** [review]

**Comment by dreispt** [comment]
Please resolve conflicts

**Comment by gdgellatly** [comment]
This PR is no longer relevant, module long since migrated.

**Pull request closed without merging**